File: scripts/create_zip.py
import zipfile
import os
import shutil

zip_path = '.build-outputs/proyecto-completo.zip'
public_zip_path = 'public/proyecto-completo.zip'
exclude_dirs = {'node_modules', '.git', '.gradle', 'build', '.build-outputs'}

os.makedirs('.build-outputs', exist_ok=True)
os.makedirs('public', exist_ok=True)

with zipfile.ZipFile(zip_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
    for root, dirs, files in os.walk('.'):
        dirs[:] = [d for d in dirs if d not in exclude_dirs and not d.startswith('.')]
        for file in files:
            if file.endswith('.zip') or file.startswith('.'):
                continue
            full_path = os.path.join(root, file)
            rel_path = os.path.relpath(full_path, '.')
            zipf.write(full_path, rel_path)

shutil.copyfile(zip_path, public_zip_path)

apk_source = '.build-outputs/app-debug.apk'
if os.path.exists(apk_source):
    shutil.copyfile(apk_source, 'public/app-debug.apk')

print('ZIP package ready:', zip_path, f"({os.path.getsize(zip_path)} bytes)")
print('Public ZIP ready:', public_zip_path, f"({os.path.getsize(public_zip_path)} bytes)")

File: scripts/create_zip.js
import fs from 'node:fs';
import path from 'node:path';
import { spawnSync } from 'node:child_process';
import { fileURLToPath } from 'node:url';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const ROOT_DIR = path.resolve(__dirname, '..');

const zipDestDir = path.join(ROOT_DIR, '.build-outputs');
const publicDir = path.join(ROOT_DIR, 'public');
const zipPath = path.join(zipDestDir, 'proyecto-completo.zip');
const publicZipPath = path.join(publicDir, 'proyecto-completo.zip');
const apkSource = path.join(zipDestDir, 'app-debug.apk');
const publicApk = path.join(publicDir, 'app-debug.apk');

if (!fs.existsSync(zipDestDir)) {
  fs.mkdirSync(zipDestDir, { recursive: true });
}
if (!fs.existsSync(publicDir)) {
  fs.mkdirSync(publicDir, { recursive: true });
}

// Copy APK if present
if (fs.existsSync(apkSource)) {
  fs.copyFileSync(apkSource, publicApk);
  console.log(`[ZIP] Synchronized APK to ${publicApk}`);
} else if (fs.existsSync(publicApk)) {
  fs.copyFileSync(publicApk, apkSource);
}

// Try running python3 create_zip.py if python3 is available
let pySuccess = false;
try {
  const pyCheck = spawnSync('python3', ['scripts/create_zip.py'], {
    cwd: ROOT_DIR,
    stdio: 'inherit'
  });
  if (pyCheck.status === 0) {
    pySuccess = true;
  }
} catch {
  pySuccess = false;
}

if (!pySuccess) {
  // If python3 not available, ensure at least existing ZIPs are synced
  if (fs.existsSync(publicZipPath) && !fs.existsSync(zipPath)) {
    fs.copyFileSync(publicZipPath, zipPath);
  } else if (fs.existsSync(zipPath) && !fs.existsSync(publicZipPath)) {
    fs.copyFileSync(zipPath, publicZipPath);
  }
  console.log('[ZIP] ZIP assets checked and verified without Python.');
}
